lib/model-builder.js

```diff
@@ -332,6 +339,38 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     dataSource.attach(this);
   };
 
+  // initialize getMergePolicy if not done already via model heritance
+  ModelClass.getMergePolicy = ModelClass.getMergePolicy || function() {
+    const mergePolicy = {
+      description: {replace: true}, // string or array
+      options: {patch: true}, // object
+      properties: {patch: true}, // object
+      hidden: {replace: false}, // array
+      protected: {replace: false}, // array
+      indexes: {patch: true}, // object
+      methods: {patch: true}, // object
+      mixins: {patch: true}, // object
+      // validations: {patch: true}, // object // NOTE: not implemented
+      relations: {patch: true},
+      scope: {replace: true}, // object
+      scopes: {patch: true}, // object
+      acls: {rank: true}, // array
+      // this setting controls which child model property's value allows deleting
+      // a base model's property
+      __delete: null,
+      // this setting controls the default merge behaviour for settings not defined
+      // in the mergePolicy specification
+      __default: {replace: true},
+    };
+
+    // make sure we take settings of current model with `this`
+    // model might have no settings in case it is an instance of builtin ModelBaseClass
+    return (this.settings || {}).mergePolicy || mergePolicy;
+  };
+
+  // ModelClass.base might not have getMergePolicy() if it is an instance of builtin ModelBaseClass
+  ModelClass.base.getMergePolicy = ModelClass.base.getMergePolicy || ModelClass.getMergePolicy;
```

Comment:
Let's say we had three models, a parent `A` and two children `B` and `C`.
With this line, if `A` had no instance of `.getMergePolicy`, then whichever of the children has its `.getMergePolicy` called first will alter `A` to share the same policy. If `B` and `C` each have different merge policies, would this not have unintended side-effects for one of the two children?